TsRandomizer/LevelObjects/TextReplacer.cs

```diff
@@ -1,63 +1,131 @@
+                                "'Kobo'... There's no one in our village by that name... I don't think this 'memory' is real.");
+                        }
+                    };
+                case "11.4":
+                    return () =>
+                    {
+                        if (options.GyreArchives)
+                            TimeSpinnerGame.Localizer.OverrideKey("q_ram_4_lun_29alt",
+                                "It says, 'Redacted Temporal Research: Lord of Ravens'. Maybe I should ask the crow about this...");
+                    };
+                //case "3.16":
+                //    return () =>
+                //    {
+                //        TimeSpinnerGame.Localizer.OverrideKey("sign_forest_directions",
+                //            GetRequiredProgressionHint(itemLocations, level.ID, level.RoomID));
+                //    };
+                default: return () => { };
+            }
+        }
+
+        internal static string GetProgressionHint(ItemLocationMap itemLocations, int levelId, int roomId)
+        {
+            List<ItemLocation> locations = (List<ItemLocation>)itemLocations.ToList(typeof(ItemLocation));
+            List<ItemLocation> requiredLocations = locations.FindAll(l => l.ItemInfo.IsProgression);
+            ItemLocation requiredLocation = requiredLocations.PopRandom(new Random(levelId * 100 + roomId)); //same hint every time per location
+            string hint = string.Format("{0} holds something useful.", requiredLocation.Name);
+            return hint;
+        }
+
+        internal static string GetRandomItemHint(ItemLocationMap itemLocations, int levelId, int roomId)
+        {
+            //needs fancying up to look good, but it works
+            List<ItemLocation> locations = (List<ItemLocation>)itemLocations.ToList(typeof(ItemLocation));
+            ItemLocation hintLocation = locations.PopRandom(new Random(levelId * 100 + roomId));
+            string hint = string.Format("{0} holds {1}.", hintLocation.Name, hintLocation.ItemInfo.Identifier.ToString());
+            return hint;
+        }
+
+        internal static string GetRequiredItemHint(ItemLocationMap itemLocations, int levelId, int roomId)
+        {
+            List<ItemLocation> locations = (List<ItemLocation>)itemLocations.ToList(typeof(ItemLocation));
+            var explicitlyRequiredItemLocations = itemLocations.Where(l => l.ItemInfo.IsExplicitlyRequired);
+            var randomRequiredItemLocation = explicitlyRequiredItemLocations.ToList().PopRandom(new Random(levelId * 100 + roomId));
+            string hint = string.Format("The path to the Ancient Pyramid goes through {0} {1}", randomRequiredItemLocation.Name, randomRequiredItemLocation.AreaName);
+            return hint;
+        }
+
+        internal static string GetRequiredProgressionHint(ItemLocationMap itemLocations, int levelId, int roomId)
+        {
+            int randomSeed = levelId * 100 + roomId;
```

Comment:
so this seeding is a bit off, as that means that a certain room wil always select element x in the explicitlyRequiredItemLocations.ToList()
the seed should be based on the rando seed so its different each run